java/ql/lib/semmle/code/java/dataflow/RangeUtils.qll

```diff
@@ -86,8 +87,29 @@ private predicate constantIntegerExpr(Expr e, int val) {
     arrlen.getQualifier() = a.getAnAccess() and
     e = arrlen
   )
+  or
+  CalcConstants::calculateIntValue(e) = val
+}
+
+pragma[nomagic]
+private predicate constantBooleanExpr(Expr e, boolean val) {
+  e.(CompileTimeConstantExpr).getBooleanValue() = val
+  or
+  exists(SsaExplicitUpdate v, Expr src |
```

Comment:
I see that you include SSA steps here; why not add that inside the parameterized module?

Reply:
See above.